### DISCUSSION TITLE
SMS on keycloak 17.0.1

### DISCUSSION THREAD

**Discussion by jht-prog** [dasniko/keycloak-2fa-sms-authenticator#14] [Q&A]
Will this code work with keycloak 17.0.1? Can you specify where the java file has to go, it looks like the providers directory? Any other file changes that need to be made for this version?

thanks in advance.....

**Comment by dasniko** [comment]
As you can see in the current [`pom.xml`](https://github.com/dasniko/keycloak-2fa-sms-authenticator/blob/5b8e26cfeaffb0bfcb99ad8c3819c4f37a8d80cc/pom.xml#L15), it's already on Keycloak 17.
For deployment to Keycloak, please refer to the official Keycloak docs. In most cases, copying the generated fat-jar into the `/providers` directory should be enough.

**Comment by jht-prog** [comment]
Thank you for the response and the instructions.

Sent from the all new AOL app for Android 
 
  On Fri, Apr 8, 2022 at 11:22 AM, Niko ***@***.***> wrote:   


As you can see in the current pom.xml, it's already on Keycloak 17.
For deployment to Keycloak, please refer to the official Keycloak docs. In most cases, copying the generated fat-jar into the /providers directory should be enough.

**Comment by jht-prog** [comment]
Can you provide the recipe for building the jar file?

**Comment by dasniko** [comment]
`mvn clean package` !?
C'mon, it's not that hard, it's a regular Maven-Java project.

Please note that this is just an example project, there's no support available!

**Comment by jht-prog** [comment]
Thank-you for your patience with me. Things are close now but getting this error.

2022-04-11 22:58:31,752 WARN  [org.keycloak.services] (executor-thread-10) KC-SERVICES0013: Failed authentication: java.lang.NullPointerException: Cannot invoke "org.keycloak.models.AuthenticatorConfigModel.getConfig()" because "config" is null

**Comment by jht-prog** [comment]
Disregard my last email. I believe I know what the problem is.

Sent from the all new AOL app for Android 
 
  On Mon, Apr 11, 2022 at 1:51 PM, Niko ***@***.***> wrote:   


mvn clean package !?
C'mon, it's not that hard, it's a regular Maven-Java project.

Please note that this is just an example project, there's no support available!
